src/platform/plugins/shared/navigation/public/solution_tour/solution_tour.ts

```diff
@@ -29,22 +29,43 @@ export class SolutionNavigationTourManager {
   ) {}
 
   async startTour(): Promise<void> {
-    // first start the spaces tour (if applicable)
-    if (this.deps.spacesSolutionViewTourManager) {
-      const spacesTour = await this.deps.spacesSolutionViewTourManager.startTour();
-      if (spacesTour.result === 'started') {
-        await this.deps.spacesSolutionViewTourManager.waitForTourEnd();
-      }
+    const { getTourQueueStateManager, TOURS } = await import('@kbn/tour-queue');
+    const tourQueueStateManager = getTourQueueStateManager();
+    // Register and get cleanup function
+    const removeTour = tourQueueStateManager.registerTour(TOURS.NAVIGATION);
+    const shouldShow = tourQueueStateManager.shouldShowTour(TOURS.NAVIGATION);
+    if (!shouldShow) {
```

Comment:
Is there a race condition here with other tours? Could it be that the security tour is registered first and shown first while this one hasn't had a chance to run yet? I think async loading for tour code might hurt us here and increase chances of the bug.

So Is it guaranteed, or do we just hope that this never happens?

Just checking - I'm not sure what's better: async loading versus sync loading. Async is faster, but sync is more reliable for avoiding race conditions.